svf-llvm/lib/SVFIRBuilder.cpp

```diff
@@ -936,7 +936,17 @@ void SVFIRBuilder::visitBranchInst(BranchInst &inst)
     BranchStmt::SuccAndCondPairVec successors;
     for (u32_t i = 0; i < inst.getNumSuccessors(); ++i)
     {
-        const Instruction* succInst = &inst.getSuccessor(i)->front();
+        BasicBlock* succBB = inst.getSuccessor(i);
+        Instruction* succInst = nullptr;
+        for (auto& inst : *succBB)
+        {
+            if (not isIntrinsicInst(&inst))
+            {
+                succInst = &inst;
+                break;
+            }
+        }
+        assert(succInst && "no non-intrinsic instruction in the successor BB");
```

Comment:
Better to find the next basic block if this bb has intrinsic instructions only before this assertion is made

Reply:
I forgot whether we have built ICFG yet or not if built we would use the ICFG’s edge to find the successor instructions

Reply:
> Better to find the next basic block if this bb has intrinsic instructions only before this assertion is made

But this can be hard when succBB has more than 1 successor

Reply:
If succBB has only one successor, I think it's ok